rfcs/README.md

```diff
@@ -10,26 +10,30 @@ them easier to comment on.
 
 The following is the RFC process for Openverse:
```

Comment:
I wonder if we can get rid of this section (or others in this document). Most of it repeats (in vaguer terms) what is described much more clearly in the new project process document.

Reply:
Dramatically simplified this README, let me know what you think.

Reply:
Fantastic :+1: